### ISSUE TITLE
`Microsoft.Extensions.Caching.StackExchangeRedis` not working with IPv6 because using an old version of `StackExchange.Redis`

### ISSUE THREAD

**Issue by mathieu-benoit** [dotnet/aspnetcore#45424]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When using `Microsoft.Extensions.Caching.StackExchangeRedis` on a single-stack IPv6 cluster, we are unable to communicate with a redis database:
```
Can't access cart storage. StackExchange.Redis.RedisConnectionException: It was not possible to connect to the redis server(s). UnableToConnect on redis-cart:6379/Interactive, Initializing/NotStarted, last: NONE, origin: BeginConnectAsync, outstanding: 0, last-read: 0s ago, last-write: 0s ago, keep-alive: 60s, state: Connecting, mgr: 10 of 10 available, last-heartbeat: never, global: 252s ago, v: 2.2.4.27433
   at StackExchange.Redis.ConnectionMultiplexer.ConnectImplAsync(ConfigurationOptions, TextWriter ) in /_/src/StackExchange.Redis/ConnectionMultiplexer.cs:line 879
   at Microsoft.Extensions.Caching.StackExchangeRedis.RedisCache.ConnectAsync(CancellationToken )
   at Microsoft.Extensions.Caching.StackExchangeRedis.RedisCache.GetAndRefreshAsync(String, Boolean, CancellationToken )
   at Microsoft.Extensions.Caching.StackExchangeRedis.RedisCache.GetAsync(String, CancellationToken )
   at cartservice.cartstore.RedisCartStore.GetCartAsync(String) in /app/cartstore/RedisCartStore.cs:line 104
```
It turns out (like described [here](https://github.com/GoogleCloudPlatform/microservices-demo/pull/1340#issuecomment-1335303056)) that `Microsoft.Extensions.Caching.StackExchangeRedis` is using a very old version of [`StackExchange.Redis`](https://github.com/StackExchange/StackExchange.Redis): [2.2.4](https://github.com/StackExchange/StackExchange.Redis/releases/tag/2.2.4) - defined [here](https://github.com/dotnet/aspnetcore/blob/main/eng/Versions.props#L293). But in [2.6.62](https://github.com/StackExchange/StackExchange.Redis/issues/2240#issuecomment-1236203477) this issue was fixed in `StackExchange.Redis`. Is there any plan to upgrade that old 2.2.4 to a [most recent version](https://github.com/StackExchange/StackExchange.Redis/tags) in `Microsoft.Extensions.Caching.StackExchangeRedis`?

### Expected Behavior

`Microsoft.Extensions.Caching.StackExchangeRedis` should work on IPv6 stack.

### Steps To Reproduce

You need to have a single-stack IPv6 cluster.

### Exceptions (if any)

```
Can't access cart storage. StackExchange.Redis.RedisConnectionException: It was not possible to connect to the redis server(s). UnableToConnect on redis-cart:6379/Interactive, Initializing/NotStarted, last: NONE, origin: BeginConnectAsync, outstanding: 0, last-read: 0s ago, last-write: 0s ago, keep-alive: 60s, state: Connecting, mgr: 10 of 10 available, last-heartbeat: never, global: 252s ago, v: 2.2.4.27433
   at StackExchange.Redis.ConnectionMultiplexer.ConnectImplAsync(ConfigurationOptions, TextWriter ) in /_/src/StackExchange.Redis/ConnectionMultiplexer.cs:line 879
   at Microsoft.Extensions.Caching.StackExchangeRedis.RedisCache.ConnectAsync(CancellationToken )
   at Microsoft.Extensions.Caching.StackExchangeRedis.RedisCache.GetAndRefreshAsync(String, Boolean, CancellationToken )
   at Microsoft.Extensions.Caching.StackExchangeRedis.RedisCache.GetAsync(String, CancellationToken )
   at cartservice.cartstore.RedisCartStore.GetCartAsync(String) in /app/cartstore/RedisCartStore.cs:line 104
```

### .NET Version

`7.0.100`

### Anything else?

- Using [`Microsoft.Extensions.Caching.StackExchangeRedis` `7.0.0`](https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/src/cartservice/src/cartservice.csproj)
- Used in container, [`Dockerfile` here](https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/src/cartservice/src/Dockerfile)
- Code used to call the redis database [here using `IDistributedCache`](https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/src/cartservice/src/cartstore/RedisCartStore.cs#L90)

**Comment by Tratcher** [comment]
Does it work if you add a direct dependency on StackExchange.Redis  2.6.62?

**Comment by mathieu-benoit** [comment]
I haven't tried it yet, that was my next step. I need to find how to do that.

**Comment by Tratcher** [comment]
Add it here:
https://github.com/GoogleCloudPlatform/microservices-demo/blob/229e3b0b05cef780ad6506ec936e8f7e63a5fe80/src/cartservice/src/cartservice.csproj#L10

**Comment by mathieu-benoit** [comment]
Thanks, just did https://github.com/GoogleCloudPlatform/microservices-demo/blob/mathieu-benoit/ipv6/src/cartservice/src/cartservice.csproj#L11, will test and will report back as soon as I have more info based on this.

**Comment by mathieu-benoit** [comment]
Just confirming that it's working successfully with this direct dependency update:
```
<PackageReference Include="Microsoft.Extensions.Caching.StackExchangeRedis" Version="7.0.0" />
<PackageReference Include="StackExchange.Redis" Version="2.6.80" />
```
Thanks!

Note: I don't know what you want to do with this issue, leave it to you to see for the next step to see if the `StackExchange.Redis` version should be upgraded upstream or not.

**Comment by adityamandaleeka** [comment]
Thanks @mathieu-benoit. 

cc @mgravell. Should we upgrade this dependency?

**Comment by mgravell** [comment]
Absolutely, but IMO we should combine this with reinvigorating the test suite for this scenario.

**Comment by mgravell** [comment]
fixed by https://github.com/dotnet/aspnetcore/pull/45261 (if/when that gets merged)

**Comment by mgravell** [comment]
^^^ is merged; I *believe* this improves the IPv6 story, which means this should be a little happier 👍

**Comment by mgravell** [comment]
Closed, completed by [#45261](https://github.com/dotnet/aspnetcore/pull/45261)
